src/goto-programs/lazy_goto_model.h

```diff
@@ -15,11 +15,18 @@
 class cmdlinet;
 class optionst;
 
+/// Interface
+struct can_produce_functiont
+{
+  virtual bool can_produce_function(const irep_idt &id) const = 0;
+};
+
 /// Model that holds partially loaded map of functions
-class lazy_goto_modelt
+class lazy_goto_modelt : public can_produce_functiont
```

Comment:
I don't know what a `can_produce_functiont` is supposed to be, hence I can't judge whether this is-a relation is justified.

Reply:
See new docs on the interface

Reply:
Thanks a lot for the clarification!